src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1675 @@
+
+    AccountID const thisDoor = sleB[sfAccount];
+    auto const sleDoor = [&] { return view.read(keylet::account(thisDoor)); }();
+
+    if (!sleDoor)
+    {
+        return {r, q, tecINTERNAL};  // LCOV_EXCL_LINE
+    }
+
+    auto const masterKey = [&]() -> std::optional<AccountID> {
+        if (sleDoor->isFlag(lsfDisableMaster))
+            return std::nullopt;
+        return thisDoor;
+    }();
+
+    std::optional<AccountID> regularKey = (*sleDoor)[~sfRegularKey];
+
+    auto const sleS = view.read(keylet::signers(sleB[sfAccount]));
+    if (!sleS)
+    {
+        if (masterKey || regularKey)
+        {
+            q = 1;
+            if (masterKey)
+                r[*masterKey] = std::numeric_limits<std::uint16_t>::max();
+            if (regularKey)
+                r[*regularKey] = std::numeric_limits<std::uint16_t>::max();
+
+            return {std::move(r), q, tesSUCCESS};
+        }
+        return {r, q, tecXCHAIN_NO_SIGNERS_LIST};
+    }
+    q = (*sleS)[sfSignerQuorum];
+
+    auto const accountSigners = SignerEntries::deserialize(*sleS, j, "ledger");
+
+    if (!accountSigners)
+    {
+        return {r, q, tecINTERNAL};  // LCOV_EXCL_LINE
+    }
+
+    for (auto const& as : *accountSigners)
+    {
+        r[as.account] = as.weight;
+    }
+
+    // add the master and regular keys. If they are already part of the signer's
+    // list, overwrite their weights.
+    if (masterKey)
+        r[*masterKey] = std::numeric_limits<std::uint16_t>::max();
+    if (regularKey)
+        r[*regularKey] = std::numeric_limits<std::uint16_t>::max();
+
+    return {std::move(r), q, tesSUCCESS};
+};
+}  // namespace
+//------------------------------------------------------------------------------
+
+NotTEC
+BridgeCreate::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureXChainBridge))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;  // LCOV_EXCL_LINE
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const account = ctx.tx[sfAccount];
+    auto const reward = ctx.tx[sfSignatureReward];
+    auto const minAccountCreate = ctx.tx[~sfMinAccountCreateAmount];
+    auto const bridge = ctx.tx[sfXChainBridge];
+    if (bridge.lockingChainDoor() == bridge.issuingChainDoor())
+    {
+        return temEQUAL_DOOR_ACCOUNTS;
+    }
+
+    if (bridge.lockingChainDoor() != account &&
+        bridge.issuingChainDoor() != account)
+    {
+        return temSIDECHAIN_NONDOOR_OWNER;
+    }
+
+    if (isXRP(bridge.lockingChainIssue()) != isXRP(bridge.issuingChainIssue()))
+    {
+        // Because ious and xrp have different numeric ranges, both the src and
+        // dst issues must be both XRP or both IOU.
+        return temSIDECHAIN_BAD_ISSUES;
+    }
+
+    if (!isXRP(reward) || reward.signum() <= 0)
+    {
+        return temXCHAIN_BRIDGE_BAD_REWARD_AMOUNT;
```

Comment:
I know we have different thresholds for when to add comments. I prefer not to comment code that just repeats what a simple line of code does. Things like:
```
// Add 1 to i
i = i+1;
``` 
Hides important comments because the code is cluttered with unimportant comments. I think  
``` 
// Reward may only be XRP and must be greater than zero 
if (!isXRP(reward) || reward.signum() <= 0) 
``` 
Doesn't make the code any clearer. 

I know this is a discussion we've often had. I don't think we're going to agree.